src/main/java/ch/njol/skript/structures/StructVariables.java

```diff
@@ -227,13 +231,15 @@ public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResu
 			}
 			variables.add(new NonNullPair<>(name, o));
 		}
-		getParser().getCurrentScript().addData(new DefaultVariables(variables));
+		script.addData(new DefaultVariables(variables)); // we replace the previous entry
 		return true;
 	}
 
 	@Override
 	public boolean load() {
 		DefaultVariables data = getParser().getCurrentScript().getData(DefaultVariables.class);
+		if (data == null) // band-aid fix for this section's behaviour being handled by a previous section
+			return true; // see https://github.com/SkriptLang/Skript/issues/6013
 		for (NonNullPair<String, Object> pair : data.getVariables()) {
```

Comment:
Maybe we could have some sort of marker that the variables have been loaded (maybe put it on the data)? We only need to call this method once and it'll handle all instances of StructVariable

Reply:
That seems like a nice idea with no real option for execution; we'd probably have to store it inside DefaultVariables object but then we'd need to check if it's present in the data already so it would be redundant since if the DefaultVariables thing is there then we need to deal with it anyway

Reply:
Since the data is created during preload, it will all be present at this stage (every structure has preLoad called before the first load call). Thus, there is no need for this loading step for each Structure. As you suggest, storing this inside DefaultVariables should suffice.